src/components/Modal.react.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+
+class Modal extends React.Component {
+  constructor(props){
+    super(props)
+    this.state = { open: this.props.open, message: this.props.message, progressBar: this.props.progressBar }
```

Comment:
Given that you are not modifying the state internally, you should be able to use the `props` objet directly, and not need to use `state` at all.

Reply:
:thinking: so, no need for `componentWillReceiveProps` below either? `this.props` will always have the right values?

Reply:
Indeed. If you're working with react-redux, in most cases you should not need to use state at all. The idea is that the single source of truth is the redux state (not to be confused with component state), and it is propagated to the components' props, which re-render automatically whenever their injected props change.

You should only use state if a component should have an internal state that differs (or is not reflected at all) in the global state.